Libraries/HyperV.psm1

```diff
@@ -307,21 +347,28 @@ Function CreateHyperVGroup([string]$HyperVGroupName, [string]$HyperVHost)
     return $retValue
 }
 
-Function CreateHyperVGroupDeployment([string]$HyperVGroup, $HyperVGroupNameXML, $HyperVHost, $SourceOsVHDPath, $DestinationOsVHDPath, $VMGeneration)
+Function CreateHyperVGroupDeployment([string]$HyperVGroup, $HyperVGroupNameXML, $HyperVHostArray, $SourceOsVHDPath, $DestinationOsVHDPath, $VMGeneration)
 {
     $HyperVMappedSizes = [xml](Get-Content .\XML\AzureVMSizeToHyperVMapping.xml)
     $CreatedVMs =  @()
     $OsVHD = $BaseOsVHD
-    $InterfaceAliasWithInternet = (Get-NetIPConfiguration -ComputerName $HyperVHost | where {$_.NetProfile.Name -ne 'Unidentified network'}).InterfaceAlias
-    $VMSwitches = Get-VMSwitch | where {$InterfaceAliasWithInternet -match $_.Name}
     $ErrorCount = 0
     $i = 0
-    $CurrentHyperVGroup = Get-VMGroup -Name $HyperVGroupName -ComputerName $HyperVHost
+    $CurrentHyperVGroup = Get-VMGroup -Name $HyperVGroupName -ComputerName $HyperVHostArray[$i]
+    $HyperVHost = $HyperVHostArray[$i]
     if ( $CurrentHyperVGroup.Count -eq 1)
     {
         foreach ( $VirtualMachine in $HyperVGroupXML.VirtualMachine)
         {
+            if ($VirtualMachine.DeployOnSecondaryHyperVHost -and ($TestLocation -match ",")) {
+                $hostNumber = $HyperVGroupXML.VirtualMachine.indexOf($VirtualMachine)
+                $HyperVHost = $xmlConfig.config.HyperV.Hosts.ChildNodes[$hostNumber].ServerName
+                $SourceOsVHDPath = $xmlConfig.config.HyperV.Hosts.ChildNodes[$hostNumber].SourceOsVHDPath
+                $DestinationOsVHDPath = $xmlConfig.config.HyperV.Hosts.ChildNodes[$hostNumber].DestinationOsVHDPath
+            }
             $vhdSuffix = [System.IO.Path]::GetExtension($OsVHD)
+            $InterfaceAliasWithInternet = (Get-NetIPConfiguration -ComputerName $HyperVHost | where {$_.NetProfile.Name -ne 'Unidentified network'}).InterfaceAlias
+            $VMSwitches = Get-VMSwitch | where {$InterfaceAliasWithInternet -match $_.Name}

```

Comment:
add here maybe to select only the 1st switch, in case there are 2 switches that are connected to the internet?